tests/aggregation_tests.py

```diff
@@ -11,278 +11,282 @@
         self.assertEqual(
             self.cursor.fetchall(),
             [
-                (2, 'US', 12252L, '2016-09-19', 'monthly', 1),
-                (2, 'US', 12252L, '2016-09-26', 'monthly', 2)  # grouped two entries
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'US', 12252L, 1, 0.0),  
+                (datetime.datetime(2016, 9, 20, 0, 0), 2, 'US', 12252L, 2, 0.0) # grouped two entries
             ])
-    
+
+
     def test_group_by_distinct_ip(self):
-        # GIVEN 7 entries of the same asn, risk and country from hostA (71.3.0.1) and hostB (190.81.134)
-        # First week: 2 hostA entries, 1 hostB entry
-        # Second week: 2 hostA entries, 2 hostB entries
-        ntp_scan_csv = dedent('''\
+        # GIVEN 3 entries of the same asn, risk, country an IP,
+        # two of which within same day
+        scan_csv = dedent('''\
         ts,ip,risk_id,asn,cc
-        2016-09-20T00:00:01+00:00,71.3.0.1,2,12252,US
-        2016-09-20T00:00:01+00:00,71.3.0.1,2,12252,US
-        2016-09-20T00:00:01+00:00,190.81.134.11,2,12252,US
-        2016-09-27T00:00:01+00:00,71.3.0.1,2,12252,US
-        2016-09-28T00:00:01+00:00,71.3.0.1,2,12252,US
-        2016-09-28T00:00:01+00:00,190.81.134.11,2,12252,US
-        2016-09-29T00:00:01+00:00,190.81.134.11,2,12252,US
+        2016-09-20T00:00:01+00:00,190.81.135.11,2,12252,US
+        2016-09-20T00:00:01+00:00,190.81.135.11,2,12252,US
+        2016-09-29T00:00:01+00:00,190.81.135.11,2,12252,US
         ''')
-        self.cursor.copy_expert("COPY logentry from STDIN csv header", StringIO(ntp_scan_csv))
+        self.cursor.copy_expert("COPY logentry from STDIN csv header", StringIO(scan_csv))
 
-        # WHEN grouped entries get created
-        main.create_count()
+        main.aggregate()
 
-        # THEN count table should have 2 rows corresponding to weeks, with properly grouped entries
+        # count table should have 2 entries with 1 count each
         self.cursor.execute('select * from count;')
         self.assertEqual(
             self.cursor.fetchall(),
             [
-                # First week: 2 entries from hostA count as one
-                (2, 'US', 12252L, '2016-09-19', 'monthly', 2),
-
-                # Second week: duplicated entries for hostA and hostB will merge to single one for each host
-                (2, 'US', 12252L, '2016-09-26', 'monthly', 2)
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'US', 12252L, 1, 0.0),  
+                (datetime.datetime(2016, 9, 20, 0, 0), 2, 'US', 12252L, 1, 0.0)
             ])
 
-    def test_group_by_ip_week_distinct_risk(self):
-        # GIVEN 4 entries of the same asn, week and country from hostA (71.3.0.1) and hostB (190.81.134)
-        # hostA: 2 entries of the same risk type
+
+    def test_group_by_risk(self):
+        # GIVEN 3 entries of the same asn, day and country from hostA (71.3.0.1) and hostB (190.81.134)
+        # hostA: 1 entry with risk type matching with hostB
```

Comment:
You removed one extra entry from hostA with the same risk. It could be useful here to check that such entry should be grouped. (it is somewhat tested already in the above tests for ip\day, but having it here will make this test complete in itself)